modules/mono/glue/GodotSharp/GodotSharp/Core/Bridge/ScriptManagerBridge.cs

```diff
@@ -494,16 +563,51 @@ internal static unsafe void GetOrLoadOrCreateScriptForType(Type scriptType, godo
                     if (_pathTypeBiMap.TryGetScriptPath(scriptType, out scriptPath))
                         return true;
 
+                    if (scriptType.IsConstructedGenericType)
+                    {
+                        // If the script type is generic, also try looking for the path of the generic type definition
+                        // since we can use it to create the script.
+                        Type genericTypeDefinition = scriptType.GetGenericTypeDefinition();
+                        if (_pathTypeBiMap.TryGetGenericTypeDefinitionPath(genericTypeDefinition, out scriptPath))
+                            return true;
+                    }
+
                     CreateScriptBridgeForType(scriptType, outScript);
                     scriptPath = null;
                     return false;
                 }
             }
 
+            static string GetVirtualConstructedGenericTypeScriptPath(Type scriptType, string scriptPath)
+            {
+                // Constructed generic types all have the same path which is not allowed by Godot
+                // (every Resource must have a unique path). So we create a unique "virtual" path
+                // for each type.
+
+                if (!scriptPath.StartsWith("res://"))
+                {
+                    throw new ArgumentException("Script path must start with 'res://'.", nameof(scriptPath));
+                }
+
+                scriptPath = scriptPath.Substring("res://".Length);
+                return $"csharp://{scriptPath}:{scriptType}.cs";
+            }
+
             if (GetPathOtherwiseGetOrCreateScript(scriptType, outScript, out string? scriptPath))
             {
                 // This path is slower, but it's only executed for the first instantiation of the type
 
+                if (scriptType.IsConstructedGenericType)
```

Comment:
At that point, if this condition is `true`, it is possible that `scriptPath` already contains a `csharp://` path, right? If that's the case, `GetVirtualConstructedGenericTypeScriptPath` will throw.

Reply:
That's a really good point, but I don't think it actually ever happens in practice. We only reach this code when `GetPathOtherwiseGetOrCreateScript` returns `true`, and that only happens when there is no existing script registered in `_scriptTypeBiMap`.

The first time this code executes, when the script hasn't been created yet and thus it's not in `_scriptTypeBiMap`, the virtual path (`csharp://`) will also not have been added to `_pathTypeBiMap`. At this time, the script is created and added with the virtual path to both bimaps, so this code should not be reachable again after that.

Reply:
I've seen it raised a few times when I was testing earlier. But can't reproduce reliably. I'm wondering if that might have been a compounding error from the one @zaevi reported above (that I witnessed too).

Reply:
I was wrong, if the script is freed then it can happen because the path will still be in `_pathTypeBiMap` but the script will have been removed from `_scriptTypeBiMap`. I guess I will also check here if the path starts with `csharp://` and avoid entering the if.